lib/api/multiObjectDelete.js

```diff
@@ -0,0 +1,307 @@
+import async from 'async';
+import crypto from 'crypto';
+import { parseString } from 'xml2js';
+
+import { auth, errors } from 'arsenal';
+
+import escapeForXML from '../utilities/escapeForXML';
+import bucketShield from './apiUtils/bucket/bucketShield';
+import metadata from '../metadata/wrapper';
+import services from '../services';
+import vault from '../auth/vault';
+import { isBucketAuthorized } from './apiUtils/authorization/aclChecks';
+
+
+/*
+   Format of xml request:
+   <Delete>
+       <Quiet>true</Quiet>
+       <Object>
+            <Key>Key</Key>
+            <VersionId>VersionId</VersionId>
+       </Object>
+       <Object>
+            <Key>Key</Key>
+       </Object>
+       ...
+   </Delete>
+   */
+
+
+  /*
+  Format of xml response:
+  <?xml version="1.0" encoding="UTF-8"?>
+  <DeleteResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+    <Deleted>
+      <Key>sample1.txt</Key>
+    </Deleted>
+    <Error>
+      <Key>sample2.txt</Key>
+      <Code>AccessDenied</Code>
+      <Message>Access Denied</Message>
+    </Error>
+  </DeleteResult>
+   */
+
+/**
+* formats xml for response
+* @param {boolean} quietSetting - true if xml should just include error list
+* and false if should include deleted list and error list
+* @param {object []} errorResults - list of error result objects with each
+* object containing -- key: objectName, error: arsenal error
+* @param {string []} deleted - list of object keys deleted
+* @return {string} xml string
+*/
+function _formatXML(quietSetting, errorResults, deleted) {
+    let errorXML = [];
+    errorResults.forEach(errorObj => {
```

Comment:
Instead of a `.forEach()` with a single `.push()` inside, you can use a `.map()`.